src/web/app/login-page.component.ts

```diff
@@ -0,0 +1,220 @@
+              'Login link has already been used. If not, kindly login using the same device.');
+        }
+      } else {
+        // not redirection from email login
+        this.isLoggingInWithEmail = false;
+        // page should stop loading only if it is not still logging in with Google
+        this.isPageLoading = this.isLoggingInWithGoogle;
+      }
+    });
+  }
+
+  /**
+   * Redirects to the Google login page.
+   */
+  logInWithGoogle(): Promise<void> {
+    this.isLoggingInWithGoogle = true;
+    const googleProvider = new GoogleAuthProvider();
+    googleProvider.addScope('https://www.googleapis.com/auth/userinfo.email');
+    return this.authService.logInWithRedirect(googleProvider)
+        .then(() => {
+          this.isLoggingInWithGoogle = false;
+        })
+        .catch(() => {
+          this.isLoggingInWithGoogle = false;
+          this.statusMessageService.showErrorToast('Cannot redirect to Google login page. Please try again.');
+        });
+  }
+
+  /**
+   * Sends the login email to the specified email address.
+   */
+  logInWithEmail(loginForm: FormGroup): void {
+    if (!this.captchaSiteKey) {
+      this.captchaResponse = '';
+    }
+
+    if (!this.formLogin.valid || this.captchaResponse === undefined) {
+      this.statusMessageService.showErrorToast(
+          'Please enter a valid email address and click the reCAPTCHA before submitting.');
+      return;
+    }
+
+    this.isLoggingInWithEmail = true;
+
+    this.authService.sendLoginEmail({
+      userEmail: loginForm.controls.email.value,
+      continueUrl: `${this.frontendUrl}/web/login${window.location.search}`,
+      captchaResponse: this.captchaResponse,
+    }).pipe(finalize(() => {
+      this.isLoggingInWithEmail = false;
+    })).subscribe((resp: SendLoginEmailResponse) => {
+      if (resp.isEmailSent) {
+        window.localStorage.setItem('emailForSignIn', loginForm.controls.email.value);
+        this.isLoginEmailSent = true;
+        this.isLogInWithEmail = false;
+      } else {
+        this.statusMessageService.showErrorToast(resp.message);
+        this.resetFormGroups();
+      }
+    }, (response: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(response.error.message);
+      this.resetFormGroups();
+    });
+  }
+
+  /**
+   * Resets the email and reCAPTCHA input fields in the Log In with Email form.
+   */
+  resetFormGroups(): void {
+    (this.formLogin = this.formBuilder.group({
+      email: ['', Validators.required],
+      recaptcha: [''],
+    }));
+
+    this.reloadCaptcha();
+  }
+
+  /**
+   * Reloads the reCAPTCHA widget if a non-empty site key is present.
+   */
+  reloadCaptcha(): void {
+    if (this.captchaSiteKey) {
+      this.captchaElem.reloadCaptcha();
+    }
+  }
+
+  /**
+   * Handles successful completion reCAPTCHA challenge.
+   *
+   * @param captchaResponse user's reCAPTCHA response token.
+   */
+  handleSuccess(captchaResponse: string): void {
```

Comment:
`handleSuccess` is too ambiguous. If it's about CAPTCHA, it should be mentioned in the method name.